[ADD] Manufacturing: Dropship to subcontractor

hojo-odoo opened the pull request:
This PR adds a new doc detailing how to manufacture products using the "Dropship Subcontractor on Order" route.

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/8565).

hojo-odoo commented:
Hey @StraubCreative,

This doc is ready for review.

Thanks,
hojo

StraubCreative reviewed (changes requested):
Hi @hojo-odoo 

Thanks for sending this over. I confirmed the flow as you have it in the doc, and, indeed, it works as written 👍  

I just had a few comments and questions below for you to take a look at. The only requested change is the numeric mismatch, and everything else is contextual, which we don't even need to address on this PR.

Overall, there are a few things I noticed, as well, for you to consider:
- this flow fulfills a very specific use case, moreso the [resupply vendors flow like you see on the tutorial](https://www.youtube.com/watch?v=JW53OkdLPeg&ab_channel=Odoo), and not as much the [subcontracting dropshipping tutorial that shares the same name as this doc](https://www.youtube.com/watch?v=98nqCQsG9qI&t=24s&ab_channel=Odoo).
- in fulfilling a very specific use case, I think we can be more clear about it's purpose, in laymens terms, because how we set up our routes on both the product and the components affects the flow; I think we should be very clear (to the point of exhaustion) that this is essentially an inventory resupply flow for the business (not a revenue-generating one) which involves a manufacturing route that leverages a vendor for components and a subcontractor to manufacture the item we're buying
- there is a route selection missing on the main product, which seems super important. Choosing **Buy** or **Dropship** changes the flow. In the case of what's written here, again it *really* seems like this is a resupply vendors flow, so we would choose **Buy** route for the main product.
- more screenshots I think would be *very* useful here.

hojo-odoo commented on content/applications/inventory_and_mrp/manufacturing/workflows/subcontracting_dropship.rst:
```diff
@@ -0,0 +1,147 @@
+add a new component, click :guilabel:`Add a line`, select the component in the :guilabel:`Component`
+drop-down menu, and specify the required quantity in the :guilabel:`Quantity` field.
+
+Configure Components
```
When using subcontracting, a receipt is automatically created upon confirmation of a PO, even if the Buy route is not selected.

The Dropship route needs to be enabled to dropship the product to the customer, but this also requires the use of the MTO route, which is why I'm going to cover it in the advanced workflows doc that I'll be working on this week. I'll make sure to add a link to that doc in this one when I push it.

The paragraph before this one mentions enabling the Subcontracting BoM type and specifying a contractor.

hojo-odoo commented on content/applications/inventory_and_mrp/manufacturing/workflows/subcontracting_dropship.rst:
```diff
@@ -0,0 +1,147 @@
+
+The dropship subcontractor on order workflow consists of three steps:
+
+#. Create a |PO| to purchase the finished product from the subcontractor (*subcontractor* |PO|);
```
This will also be covered in the advanced workflows doc, since it requires enabling the MTO route as well.

hojo-odoo commented on content/applications/inventory_and_mrp/manufacturing/workflows/subcontracting_dropship.rst:
```diff
@@ -0,0 +1,147 @@
+Create purchase order
+---------------------
+
+To create a *subcontractor* |PO|, navigate to :menuselection:`Purchase app --> Orders --> Purchase
```
This doc is specific to workflows where a company wants to replenish their own stock of a subcontracted product, rather than have something subcontracted to meet the needs of an SO. That will be covered in the advanced workflows doc.

hojo-odoo commented:
Hey @StraubCreative,

I've updated this doc to include more pictures, which show the differences between the subcontractor PO and the vendor PO. 

I've also added a new admonition block that explains the differences between the Dropshipping and Dropship Subcontractor on Order routes, as well as a paragraph about where to find the subcontractor's address on the vendor PO.

The rest of your suggestions will be addressed by the advanced subcontracting workflow doc, which I will be pushing this week.

Thanks,
hojo

StraubCreative commented on content/applications/inventory_and_mrp/manufacturing/workflows/subcontracting_dropship.rst:
```diff
@@ -0,0 +1,263 @@
+#. If the workflow was started by creating an |SO|, and the finished product is not dropshipped to
+   the end customer, process the delivery order once the product has been shipped to the customer.
+
+The specific number of steps depends on the reason that the subcontracted product is being purchased
+from the subcontractor.
+
+If the reason is to fulfill a specific customer order, the process starts with creating an SO, and
+ends with delivering the product to the customer, or having the subcontractor dropship it to them.
+
+If the reason is to increase quantity of stock on-hand, the process starts with creating a PO, and
+ends with receiving the product into inventory.
```
Great concluding statements to this section, covering the range of use cases here 👍

StraubCreative commented on content/applications/inventory_and_mrp/manufacturing/workflows/subcontracting_dropship.rst:
```diff
@@ -0,0 +1,263 @@
+appears at the top of the page. This is the *subcontractor* |PO|, or the |PO| created to purchase
+the subcontracted product from the subcontractor.
+
+.. note::
```
👏 👏 👏

StraubCreative commented on content/applications/inventory_and_mrp/manufacturing/workflows/subcontracting_dropship.rst:
```diff
@@ -0,0 +1,263 @@
+         button at the top of the page.
+
+In addition, an |RfQ| is created for the components that are purchased from the vendor and sent to
+the subcontractor. However, the |RfQ| **IS NOT** automatically linked to the *subcontractor* |PO|.
```
I *think* the standard for all caps in the negative is just to do the negative word (not both the positive + negative). Open to interpretation ofc 😅 

```suggestion
the subcontractor. However, the |RfQ| is **NOT** automatically linked to the *subcontractor* |PO|.
```

StraubCreative commented on content/applications/inventory_and_mrp/manufacturing/workflows/subcontracting_dropship.rst:
```diff
@@ -0,0 +1,263 @@
+receipt or dropship order is automatically created, and can be accessed from the corresponding
+:guilabel:`Receipt` or :guilabel:`Dropship` smart button that appears at the top of the |PO|.
+
+.. image:: subcontracting_dropship/subcontractor-po.png
```
Optional:
For this screenshot, can also highlight the **Deliver to** field which provides another visual queue that this is the subcontractor PO for the main manufactured product, and that what's ordered here comes back to the business after the subcontractor produces it. 

We can also turn this **image** directive into a [figure](https://documatt.com/restructuredtext-reference/element/figure.html), which allows for the use of captions to describe the image further (can just c/p alt text to make it easy).

StraubCreative commented on content/applications/inventory_and_mrp/manufacturing/workflows/subcontracting_dropship.rst:
```diff
@@ -0,0 +1,263 @@
+of the *vendor* |PO|, and a :guilabel:`Resupply` smart button appears at the top of the
+*subcontractor* |PO|.
+
+.. image:: subcontracting_dropship/vendor-po.png
```
Similar optional feedback here. Consider:
- highlighting **Deliver to** field (along with, in this case, the **Dropship Address**
- **image** --> **figure** directive

StraubCreative reviewed (approved):
Hi @hojo-odoo 

Well done! I think this round of revisions was very helpful and clearly explains both the flow and the options therein— a tough balance to pull off given how many outcomes there are across the multitude of steps 😉 

For the optional feedback below, you can follow-up in a new [IMP] PR if you so choose (for another 2-3pts). Either way, approved for merge here.

Thank you for your hard work on this!
.....
@robodoo r+

StraubCreative commented:
@fw-bot r+
